feat(interview-prep): Converting and transfering Rosetta problems (#35770)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/08-coding-interview-prep/rosetta-code/knapsack-problem-0-1.md

---
id: 5a23c84252665b21eecc7ed1
title: Knapsack problem/0-1
challengeType: 5
---

## Description
<section id='description'>

> Write a function that solves the knapsack problem that is as follows:

Reviewer: ```suggestion
The 0-1 knapsack problem is defined as follows:
```

You are given an array of objects representing items to be put in a knapsack. The objects have 3 attributes: name, weight, and value. The items need to be selected so that the total weight does not exceed the maximum weight and the value is maximized.

> The function is given the array of objects and the maximum weight as parameters. It should return the maximum total value possible.

Reviewer: ```suggestion
```

</section>

## Instructions
<section id='instructions'>

> 

Reviewer: ```suggestion
Write a function to solve the knapsack problem.  The function is given the array of objects and the maximum weight as parameters. It should return the maximum total value possible.
```
